src/eip1271.rs

```diff
@@ -0,0 +1,69 @@
+use std::{collections::BTreeMap, convert::TryFrom};
+
+use ethers::{
+    abi::{Abi, Function, Param, ParamType, StateMutability},
+    contract::{AbiError, Contract},
+    prelude::*,
+};
+
+use crate::VerificationError;
+
+const METHOD_NAME: &str = "isValidSignature";
+
+pub async fn verify_eip1271(
+    address: [u8; 20],
+    message_hash: &[u8; 32],
+    signature: &[u8],
+) -> Result<bool, VerificationError> {
+    #[allow(deprecated)]
+    let abi = Abi {
+        constructor: None,
+        functions: BTreeMap::from([(
+            "isValidSignature".to_string(),
+            vec![Function {
+                name: "isValidSignature".to_string(),
+                inputs: vec![
+                    Param {
+                        name: " _message".to_string(),
+                        kind: ParamType::FixedBytes(32),
+                        internal_type: Some("bytes32".to_string()),
+                    },
+                    Param {
+                        name: " _signature".to_string(),
+                        kind: ParamType::Bytes,
+                        internal_type: Some("bytes".to_string()),
+                    },
+                ],
+                outputs: vec![Param {
+                    name: "".to_string(),
+                    kind: ParamType::FixedBytes(4),
+                    internal_type: Some("bytes4".to_string()),
+                }],
+                constant: None,
+                state_mutability: StateMutability::View,
+            }],
+        )]),
+        events: BTreeMap::new(),
+        errors: BTreeMap::new(),
+        receive: false,
+        fallback: false,
+    };
+
+    let client = Provider::<Http>::try_from("https://cloudflare-eth.com")
```

Comment:
Should we expose this as a configurable parameter?

Reply:
Yes, coming in a follow-up commit/PR with the config object